src/doc/src/reference/unstable.md

```diff
@@ -237,6 +238,22 @@ This can also be specified in `.cargo/config.toml` files.
 artifact-dir = "out"
 ```
 
+## build-dir
+* Original Issue: [#14125](https://github.com/rust-lang/cargo/issues/14125)
+* Tracking Issue: [#14125](https://github.com/rust-lang/cargo/issues/14125)
+
+This feature allows you to specify the directory where intermediate build artifacts will be stored.
+The final artifacts will also be contained in this directory but will be hardlinked into the artifact directory (usually `target`)
+
+This can be specified in `.cargo/config.toml` files using the `build.build-dir`.
+
+Example:
+
+```toml
+[build]
+build-dir = "out"
+```
```

Comment:
Could you include documentation in this that mirrors what would be in https://doc.rust-lang.org/cargo/reference/config.html